atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -449,6 +453,45 @@ def _get_replacement_pullspecs_OSBS_resolution(self, pullspecs):
 
         return replacements
 
+    def _replace_other_metadata_from_csv_modifications(self, operator_manifest):
+        """
+        WILL add/append/update/replace metadata in the operator manifest
+        passed in
+
+        :param operator_manifest: an OperatorManifest, as created by `from_directory()`
+        :type operator_manifest: instance of OperatorManifest
+        :return: None
+        """
+
+        def do_action(operator_manifest, action):
+            mods = self._fetch_operator_csv_modifications()
+            if (mods.get(action)):
+                # Loop over mod data and apply action
+                for action_field in mods[action]:
+                    for item in mods[action][action_field]:
+                        if action == 'append':
+                            if isinstance(mods[action][action_field][item], list):
+                                # Append to existing list
+                                if operator_manifest.csv.data[action_field].get(item):
```

Comment:
@mprahl @MartinBasti 
is it possible that freshmaker will pass:
append : spec->skips->['some']

but in CSV even 'spec' doesn't exist?

if yes we should also first check if even : operator_manifest.csv.data[action_field] exists (like in update)

Reply:
@rcerven; I could be wrong, but I believe that the top-level 'spec' is required by our schema check. Regardless, I've since refactored and it's significantly more robust.

Reply:
ok I've 'spec' & 'metadata' are required in schema so we are good